tools/bin/ci_integration_workflow_launcher.py

```diff
@@ -0,0 +1,185 @@
+    )
+
+
+def get_run_uuid(jobs):
+    """
+    This function relies on assumption that the first step of the first job
+
+    - name: UUID ${{ github.event.inputs.uuid }}
+      run: true
+    """
+    if jobs and len(jobs[0]["steps"]) >= 2:
+        name = jobs[0]["steps"][1]["name"]
+        m = re.match(RUN_UUID_REGEX, name)
+        if m:
+            return m.groups()[0]
+
+
+def get_response(url_or_path, params=None):
+    url = urljoin(API_URL, url_or_path)
+    response = requests.get(url, params=params, headers={"Authorization": "Bearer " + GITHUB_TOKEN})
+    response.raise_for_status()
+    return response
+
+
+def get_response_json(url_or_path, params=None):
+    response = get_response(url_or_path, params=params)
+    return response.json()
+
+
+def get_workflow_id(owner, repo, path):
+    response_json = get_response_json(f"/repos/{owner}/{repo}/actions/workflows")
+    for workflow in response_json["workflows"]:
+        if workflow["path"] == path:
+            return workflow["id"]
+
+
+def workflow_dispatch(owner, repo, workflow_id, connector):
+    run_uuid = str(uuid.uuid4())
+    url = urljoin(API_URL, f"/repos/{owner}/{repo}/actions/workflows/{workflow_id}/dispatches")
+    response = requests.post(
+        url, headers={"Authorization": "Bearer " + GITHUB_TOKEN}, json={"ref": BRANCH, "inputs": {"connector": connector, "uuid": run_uuid}}
+    )
+    response.raise_for_status()
+    return run_uuid
+
+
+def get_connector_names():
+    process = subprocess.run(["./gradlew", "integrationTest", "--dry-run"], check=True, capture_output=True, universal_newlines=True)
+
+    res = []
+    for line in process.stdout.splitlines():
+        parts = line.split(":")
+        if (
+            len(parts) >= 4
+            and parts[1] == "airbyte-integrations"
+            and parts[2] in ["connectors", "bases"]
+            and parts[-1] == "integrationTest SKIPPED"
+        ):
+            res.append(parts[3])
+    return res
+
+
+def iter_workflow_runs(owner, repo, per_page=100):
+    path = f"/repos/{owner}/{repo}/actions/runs"
+    page = None
+    while True:
+        params = {"per_page": per_page}
+        if page:
+            params["page"] = page
+        response = get_response(path, params=params)
+        response_json = response.json()
+        for workflow_run in response_json["workflow_runs"]:
+            yield workflow_run
+        if "next" not in response.links:
+            break
+        page = dict(parse_qsl(urlparse(response.links["next"]["url"]).query))["page"]
+
+
+def search_failed_workflow_runs(owner, repo, workflow_id, run_uuids):
+    run_uuids = set(run_uuids)
+    now = datetime.datetime.utcnow()
+    res = set()
+    for workflow_run in iter_workflow_runs(owner, repo):
+        if not run_uuids:
+            break
+
+        created_at = datetime.datetime.strptime(workflow_run["created_at"], "%Y-%m-%dT%H:%M:%SZ")
+        period = now - created_at
+        if period.seconds > 10800:
```

Comment:
no problem with this timer length, but any logic behind the choice?

Reply:
We scan all workflows backwards until we find all needed by uuid
This 3 hours is just additinal protection (no any reason just 3 hours :)

If we did not  found all needed workflows by uuid for some reason
just break loop when we starting to reach "old" (older 3 hours ) workflows